ui/app/contexts/metametrics.new.js

```diff
@@ -92,16 +92,25 @@ export function MetaMetricsProvider ({ children }) {
       const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
       const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
       const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
-      if (
-        match &&
+      // Start by checking for a missing match route. If this falls through to the else if, then we know we
+      // have a matched route for tracking.
+      if (!match) {
+        // We have more specific pages for each type of transaction confirmation
+        // The user lands on /confirm-transaction first, then is redirected based on
+        // the contents of state.
+        if (location.pathname !== '/confirm-transaction') {
+          // Otherwise we are legitimately missing a matching route
+          // Ideally we'd include the details of match and previousMatch here, but breadcrumbs provides
+          // a fair bit of details that should be useful.
+          captureMessage(`Segment page tracking found unmatched route`, 'info')
+        }
+      } else if (
         previousMatch.current !== match.path &&
-        // If we're in a popup or notification we don't want the initial home route to track
-        !(
-          (environmentType === 'popup' || environmentType === 'notification') &&
-          match.path === '/' &&
-          previousMatch.current === undefined
-        )
+        (environmentType !== 'notification' || match.path !== '/' || previousMatch.current !== undefined)
```

Comment:
This was challenging for me to interpret - could we maybe move the negation outside of these parens?

```suggestion
        !(environmentType === 'notification' && match.path === '/' && previousMatch.current === undefined)
```

Reply:
Sounds good to me! thanks :)